renderdoc/core/image_viewer.cpp

```diff
@@ -748,15 +749,28 @@ void ImageViewer::RefreshFile()
       return;
     }
 
-    datasize = size_t(texDetails.width) * size_t(texDetails.height) * 4 * sizeof(float);
-    data = (byte *)malloc(datasize);
-
-    if(!data)
+    size_t totalSize = size_t(texDetails.width) * size_t(texDetails.height) * 4 * sizeof(float);
+    int mipCount = 1;
+    read_data.subresources.push_back({0, totalSize});
+    // Support mipmaps in EXR, if the tile rounding mode matches the graphics APIs (rounding
+    // down for odd sizes).
+    if(exrImage.images == NULL && exrImage.tiles != NULL &&
+       exrHeader.tile_level_mode == TINYEXR_TILE_MIPMAP_LEVELS &&
+       exrHeader.tile_rounding_mode == TINYEXR_TILE_ROUND_DOWN)
     {
-      SET_ERROR_RESULT(m_Error, ResultCode::OutOfMemory,
-                       "Allocation for %zu bytes failed for EXR data", datasize);
-      return;
+      const EXRImage *exrLevel = exrImage.next_level;
+      while(exrLevel != NULL)
+      {
+        size_t mipSize = size_t(exrLevel->width) * size_t(exrLevel->height) * 4 * sizeof(float);
+        read_data.subresources.push_back({totalSize, mipSize});
+        totalSize += mipSize;
+        mipCount++;
+        exrLevel = exrLevel->next_level;
+      }
     }
+    read_data.mips = mipCount;
+    texDetails.mips = mipCount;
```

Comment:
Can you set the other members of `read_data` here? At the moment they shouldn't be used but I'd rather have them initialised properly since they're simple to get at, in case they're used in future as this path is very unlikely to be tested often.